packages/network-controller/tests/provider-api-tests/shared-tests.ts

```diff
@@ -0,0 +1,391 @@
+        { name: 'eth_feeHistory', numberOfParameters: 3 },
+        { name: 'debug_getRawHeader', numberOfParameters: 1 },
+        { name: 'debug_getRawBlock', numberOfParameters: 1 },
+        { name: 'debug_getRawTransaction', numberOfParameters: 1 },
+        { name: 'debug_getRawReceipts', numberOfParameters: 1 },
+        { name: 'debug_getBadBlocks', numberOfParameters: 0 },
+      ];
+      notHandledByMiddleware.forEach(({ name, numberOfParameters }) => {
+        describe(`method name: ${name}`, () => {
+          testsForRpcMethodNotHandledByMiddleware(name, {
+            providerType,
+            numberOfParameters,
+          });
+        });
+      });
+    });
+
+    describe('methods handled by middleware that returns a static result', () => {
+      const notHandledByMiddleware = [
+        { name: 'eth_accounts', numberOfParameters: 0, result: [] },
+        { name: 'eth_coinbase', numberOfParameters: 0, result: null },
+        { name: 'eth_hashrate', numberOfParameters: 0, result: '0x00' },
+        { name: 'eth_mining', numberOfParameters: 0, result: false },
+      ];
+      notHandledByMiddleware.forEach(({ name, numberOfParameters, result }) => {
+        describe(`method name: ${name}`, () => {
+          testsForRpcMethodWithStaticResult(name, {
+            providerType,
+            numberOfParameters,
+            result,
+          });
+        });
+      });
+    });
+
+    describe('methods with block hashes in their result', () => {
+      const methodsWithBlockHashInResponse = [
+        { method: 'eth_getTransactionByHash', numberOfParameters: 1 },
+        { method: 'eth_getTransactionReceipt', numberOfParameters: 1 },
+      ];
+
+      methodsWithBlockHashInResponse.forEach(
+        ({ method, numberOfParameters }) => {
+          describe(`method name: ${method}`, () => {
+            testsForRpcMethodsThatCheckForBlockHashInResponse(method, {
+              providerType,
+              numberOfParameters,
+            });
+          });
+        },
+      );
+    });
+
+    describe('methods that assume there is no block param', () => {
+      const assumingNoBlockParam = [
+        // { name: 'eth_getFilterLogs', numberOfParameters: 1 },
+        // { name: 'eth_blockNumber', numberOfParameters: 0 },
+        { name: 'eth_estimateGas', numberOfParameters: 2 },
+        { name: 'eth_gasPrice', numberOfParameters: 0 },
+        { name: 'eth_getBlockByHash', numberOfParameters: 2 },
+        {
+          name: 'eth_getBlockTransactionCountByHash',
+          numberOfParameters: 1,
+        },
+        {
+          name: 'eth_getTransactionByBlockHashAndIndex',
+          numberOfParameters: 2,
+        },
+        { name: 'eth_getUncleByBlockHashAndIndex', numberOfParameters: 2 },
+        { name: 'eth_getUncleCountByBlockHash', numberOfParameters: 1 },
+      ];
+
+      // NOTE: these methods do take a block param
+      // but this is not handled by our cache middleware currently
```

Comment:
That's what I was trying to convey with "assuming there is no block param". If a method doesn't take a block param, great; if it does, then the middleware ignores it, as if it doesn't take one. We can keep this for now, but I think we should clarify this in a future update, maybe by changing the variable above from `assumingNoBlockParam` to `takesNoBlockParam` or something like that.